src/main/java/org/apache/commons/collections4/bloomfilter/IndexProducer.java

```diff
@@ -117,11 +120,60 @@ public boolean test(long word) {
      * @return An int array of the data.
      */
     default int[] asIndexArray() {
-        final BitSet result = new BitSet();
+        class Indices {
```

Comment:
Note: This method now avoids the BitSet. As such I think we can remove the javadoc warning that the default implementation is slow. The note about returning in unique values in order is wrong. The default implementation now returns an array using the same order and cardinality as the `forEachIndex` method.